ddp-workspace/projects/ddp-singular/src/app/components/pages/participant-list/participant-list.component.ts

```diff
@@ -131,21 +132,19 @@ export class ParticipantsListComponent implements OnInit {
   }
 
   onEditActivity(participantGuid: string, activityToEdit: ActivityInstance): void {
+    this.isPageBusy = true;
     this.setParticipant(participantGuid);
-
-    this.activityService
-      .createInstance(this.config.studyGuid, activityToEdit.activityCode)
-      .pipe(take(1))
-      .subscribe(activity => {
-        this.setCurrentActivity(activity as ActivityInstance);
-        this.redirectToSurvey(activity.instanceGuid);
-      });
+    this.setCurrentActivity(activityToEdit, false);
+    this.redirectToSurvey(activityToEdit.instanceGuid);
+    this.isPageBusy = false;
```

Comment:
Whenever something happens I am saying the page is busy (I think it might be overkill here, but at worse, it does not hurt anything).